File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/util/AppInfo.java
package com.dbottillo.mtgsearchfree.util;

import android.content.Context;
import android.content.pm.PackageManager;

public class AppInfo {

    private final Context context;

    public AppInfo(Context context) {
        this.context = context;
    }

    public long getFirstInstallTime(){
        try {
            return context.getPackageManager().getPackageInfo(context.getPackageName(), 0).firstInstallTime;
        } catch (PackageManager.NameNotFoundException ignored) {
            return -1;
        }
    }

    public long getLastUpdateTime(){
        try {
            return context.getPackageManager().getPackageInfo(context.getPackageName(), 0).lastUpdateTime;
        } catch (PackageManager.NameNotFoundException ignored) {
            return -1;
        }
    }
}

File: MTGSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/util/AppInfo.kt
package com.dbottillo.mtgsearchfree.util

import android.content.Context
import android.content.pm.PackageManager

class AppInfo(private val context: Context) {

    val firstInstallTime: Long
        get() {
            return try {
                context.packageManager.getPackageInfo(context.packageName, 0).firstInstallTime
            } catch (ignored: PackageManager.NameNotFoundException) {
                -1
            }

        }

    val lastUpdateTime: Long
        get() {
            return try {
                context.packageManager.getPackageInfo(context.packageName, 0).lastUpdateTime
            } catch (ignored: PackageManager.NameNotFoundException) {
                -1
            }

        }
}
